src/resource/solidDataset.test.ts

```diff
@@ -2168,13 +2323,37 @@ describe("createContainerInContainer", () => {
     );
   });
 
-  it("includes the final slug with the return value, normalised to the Container's origin", async () => {
+  it("includes the final slug with the return value, normalised to the target Container's origin", async () => {
+    const mockFetch = setMockOnFetch(
+      jest.fn(window.fetch),
+      mockResponse("Arbitrary response", {
+        headers: {
+          Location: "/parent-container/child-container/",
+        },
+        url: "https://some.pod/parent-container/",
+      })
+    );
+
+    const savedSolidDataset = await createContainerInContainer(
+      "https://some.pod/parent-container/",
+      {
+        fetch: mockFetch,
+      }
+    );
+
+    expect(savedSolidDataset!.internal_resourceInfo.sourceIri).toBe(
+      "https://some.pod/parent-container/child-container/"
+    );
+  });
+
+  it("includes the final slug with the return value, relative to the target Container", async () => {
     const mockFetch = setMockOnFetch(
       jest.fn(window.fetch),
-      new Response("Arbitrary response", {
+      mockResponse("Arbitrary response", {
         headers: {
-          Location: "parent-container/child-container/",
+          Location: "child-container/",
```

Comment:
So this mock does not reflect actual server behaviour now, I think:

```suggestion
          Location: "parent-container/child-container/",
```

Reply:
Oh wait, I just noticed that this is an extra test. Is the server/spec behaviour actually that it's always relative to the Container URL, but using a preceding slash to make it relevant to the origin if so desired? If so, nice fix - then we didn't process it properly before, I suppose.

Reply:
Exactly :)